### PR TITLE
Fix cloneLooksExactlySameAsOriginal test

### LINKED ISSUE #1409: `CloneMergeTest::cloneLooksExactlySameAsOriginal` fails
After improving the test runs reliability in #1389 / #1401, one test is still failing consistently: [`CloneMergeTest::cloneLooksExactlySameAsOriginal`](https://github.com/versionpress/versionpress/blob/1a21fcb11b2693b8eac44b95afd904cfde13e517/plugins/versionpress/tests/Workflow/CloneMergeTest.php#L34-L49).

When Workflow tests run separately, or before End2End tests (see https://github.com/versionpress/versionpress/issues/1400#issuecomment-4719877160), this test passes but on a site exercised by End2End tests, it fails. Upon initial exploration, @pavelevap found out that indeed the clone looks different than the original.

I'm going to implement a workaround to make the tests pass but we should return to this and fix the underlying problem.

- [ ] Update CloneMergeTest to run from a clean WP site (it's better than skipping the test entirely). @borekb
- [ ] Find out what the problem is and implement a fix. @pavelevap

### PR DESCRIPTION
**Pull request by pavelevap**
Resolves #1409 

The cause of this problem depends on #1420, this PR will be only workaround to avoid posts with different titles published at the same time.

### CODE REVIEW

**Comment by pavelevap** [comment]
After partial revert of PR #1410, tests are failing again:

https://travis-ci.org/versionpress/versionpress/builds/518248095#L1653

image

Date for "Test post for comments" has to be changed to preserve order on cloned site.

**Comment by borekb** [comment]
There was an issue with the exit code from the `run-tests.ts` script:

Screenshot 2019-04-10 at 14 30 52

This is now resolved by #1422.

**Comment by pavelevap** [comment]
OK, this PR is correctly broken now and tests are failing :-)

**Comment by pavelevap** [comment]
This is as small as possible fix to make all tests pass again, ready for discussion and review.

I started with changing `post_date` for some posts, but it would lead to many manual changes to enforce the right order. So I ended with removing predefined `post_date` in End2End tests and ensuring that there are no posts published at the same time.

File: plugins/versionpress/tests/Automation/WpAutomation.php [hunk 1]
```diff
@@ -135,11 +135,15 @@ public function uninstallVersionPress()
      *
      * @see wp_insert_post()
      * @param array $post (as wp_insert_post)
+     * @param bool $waitOneSecond Wait to ensure that there are no testing posts published at the same time.
+     *                            Currently used only for End2End tests which would fail without it, see #1420.    
      * @return int
      */
-    public function createPost(array $post)
+    public function createPost(array $post, $waitOneSecond = false)
```
Comment (borekb) [inline plugins/versionpress/tests/Automation/WpAutomation.php]: Though it's a step in the right direction, this is still possible:

```php
$post1["post_date"] = "2011-11-11 11:11:11";
$post2["post_date"] = "2011-11-11 11:11:11";

$wpAutomation->createPost($post1, true);
$wpAutomation->createPost($post2, true);

// date collision not really prevented
```  

I think the method could look something like this:

```php
public function createPost(array $post, $managePostDateAutomatically = false)
{
    if ($managePostDateAutomatically) {
        if (isset($post["post_date"])) {
            throw new Exception('post_date must not be set when managing post date automatically');
        }
        sleep(1);
    }
    // ...
}
```

The usage of `sleep` becomes an implementation detail that could change in the future, the main thing is that the method guarantees certain behavior.

**Comment by borekb** [comment]
We had a discussion about this with @pavelevap and @JanVoracek yesterday and decided that the approach from #1410 is actually preferable as it runs workflow tests from a well-known state.

It's true that running the clone & merge tests after End2End tests allowed us to uncover an important detail about VersionPress' behavior, documented in #1420, however, we shouldn't be using a combination of End2End and Workflow tests to verify that.

So I'm going to close this PR and take a look at a change done in #1410, it will probably need some minor updates in comments but overall, that should be our solution for #1409. @pavelevap, thanks for the effort anyway, it lead us to think more deeply about how tests should be written.

**Pull request closed without merging**